### ISSUE TITLE
Renaming a type parameter results in double ticks ''

### ISSUE THREAD

**Issue by bentayloruk** [ionide/ionide-vscode-fsharp#819]
### Problem

If you refactor rename a type parameter, you end up with double ticks before the name, rather than single.

![2018-05-23 11 36 04](https://user-images.githubusercontent.com/189409/40419573-a847f220-5e7d-11e8-8642-f01414905cd4.gif)

### Steps to reproduce

* Put cursor on type param.
* Hit F2.
* Change the type param name, leaving the leading `'` in place.
* Name now starts with `''` rather than `'`.

### Workaround

Remove the leading `'` when you are making the name change.

### Machine infos

* Operating system: **Darwin**
* Arch: **x64**
* VSCode: **1.23.1**
* Runtime: **.Net**
* Mono version:
```shell
Mono JIT compiler version 5.10.1.47 (2017-12/8eb8f7d5e74 Fri Apr 13 20:18:12 EDT 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(3.6.0svn-mono-master/8b1520c8aae)
	GC:            sgen (concurrent by default)
```
* MSBuild version:
```shell
Microsoft (R) Build Engine version 15.6.0.0 (xplat-master/ca830585 Wed Mar 21 09:55:27 EDT 2018) for Mono
Copyright (C) Microsoft Corporation. All rights reserved.
```

**Comment by mvsmal** [comment]
Hi. To fix this, `prepareRename` method should be implemented in `RenameProvider`. I have a draft code which checks for leading `'`. May be better to check the symbol type, though. But which type is that? I haven't found such type as TypeParameter or anything similar.

**Comment by Krzysztof-Cieslak** [comment]
Should be fixed with #890
